yarn/datadog_checks/yarn/yarn.py

```diff
@@ -234,19 +234,35 @@ def _yarn_app_metrics(self, rm_address, app_tags, addl_tags):
         """
         Get metrics for running applications
         """
+        collect_apps_all_states = self.instance.get(
+            'collect_apps_all_states', self.init_config.get('collect_app_all_states', False)
+        )
+        collect_apps_states = self.instance.get(
+            'collect_apps_states_list', self.init_config.get('collect_apps_states_list', [])
+        )
+        if collect_apps_states:
+            collect_apps_all_states = False
+
+        collect_apps_states = [state.upper() for state in collect_apps_states]
+
         metrics_json = self._rest_request_to_json(rm_address, YARN_APPS_PATH, addl_tags)
 
         if metrics_json and metrics_json.get('apps') and metrics_json['apps'].get('app') is not None:
             for app_json in metrics_json['apps']['app']:
-                tags = self._get_app_tags(app_json, app_tags) + addl_tags
-
-                if app_json['state'] == YARN_APPLICATION_RUNNING:
+                app_state = app_json['state']
+                tags = self._get_app_tags(app_json, app_tags) + addl_tags + ['state:{}'.format(app_state)]
```

Comment:
Just to avoid the 2 temporary list creations:

```suggestion
                tags = self._get_app_tags(app_json, app_tags)
                tags.extend(addl_tags)
                tags.append('state:{}'.format(app_state))
```